vpr/src/base/read_route.cpp

```diff
@@ -445,3 +469,78 @@ static std::string format_name(std::string name) {
         return nullptr;
     }
 }
+
+/* Check if there is a discontinuity in the trace
+ * There is more to this but I'll test thoroughly, my priority is not to break it */
+static bool check_dangling_branch(int x1, int y1, int x2, int y2, int prev_x1, int prev_y1, int prev_x2, int prev_y2, std::string curr_type, std::string prev_type) {
+    
+    if(prev_x1 == -1) return false;
```

Comment:
It looks like this code is checking for connectivity by comparing the RR node co-ordinates. This won't be robust in practice since it is possible to have wires which are adjacent but not connected.

I think a more robust way to do this would be to check explicitly for connectivity based on the Routing Resource (RR) graph. I'd suggest reading through the RR graph header files to learn about it and the interface.

A rough sketch of what this could look like is:
```cpp
bool rr_nodes_are_connected(RRNodeId prev_node, RRNodeId node) {
    auto& device_ctx = g_vpr_ctx.device();
    const auto& rr_graph = device_ctx.rr_nodes;

    for (RRedgeId edge : rr_graph.edge_range(prev_node)) {
        if (rr_graph.edge_sink_node(edge) == node) {
            return true;
        }
    }
    return false;
}
```